recipes/pywr/meta.yaml

```diff
@@ -0,0 +1,72 @@
+{% set name = "pywr" %}
+{% set version = "1.0.0" %}
+{% set sha256 = "fdaddaf772c128f6bcc61e937b18dfc0ae5f80c3a0bf592fac97dd6579cd002b" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py<36]
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  host:
+    - python
+    - pip
+    - {{ compiler('c') }}
+    - numpy
+    - cython
+    - packaging
+    - glpk
+    - lp_solve
+    - future
+    - packaging
+    - setuptools
+    - setuptools_scm
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+    - pandas
+    - networkx
+    - glpk
+    - lp_solve
+    - pytables
+    # Fix pytables on windows. See https://github.com/conda-forge/pytables-feedstock/issues/31#issuecomment-410695814
+    - snappy  # [win]
+    - xlrd
+    - scipy
+    - future
+    - packaging
+    - jupyter
+    - matplotlib
```

Comment:
This doesn't stop this from being merged but should there be any version requirements on the runtime dependencies?

Reply:
Generally we have our CI setup to test against the latest versions of these libraries.

One or two could be considered optional dependencies I guess, how should those be dealt with here? I copied the list from our CI where we they are installed in order to run the tests in full, but they don't necessarily need to be installed in all cases.

Reply:
For creating new environments this will work fine.

The issue arises when someone installs `pywr` into a pre-existing environment which contains old versions of these packages. If no versions are specified the packages won't be updated as conda thinks any version of `matplotlib`/`jupyter`/... is appropriate. This will also affect installing with pip.

In my opinion the best solution is to use two jobs in your CI, one tests the latest releases and one tests the minimum supported versions. You can then adjust the minimum versions when the CI fails or modify the code to avoid the issue.

> One or two could be considered optional dependencies I guess, how should those be dealt with here?

They can be listed as "required" dependencies if you want them to be installed by default.

Reply:
Though this is all just advice for the future 😄

I'm happy to merge this if you are?

Reply:
Yeah sure. Thanks. I'll make a note of that re the CI and dependency versions.